### ISSUE TITLE
Graph/network analysis of department

### ISSUE THREAD

**Issue by redshiftzero** [lucyparsons/OpenOversight#527]
Suggested by @dismantl: a nice way to view command structure in a police department would be a visualization that displays a graph of all officers, where each node in a graph is an individual officer, and the edges represent a command relationship. 

(this is a good issue for anyone interested in d3.js)

**Comment by tomx4096** [comment]
This could be visually cool, but would it just be based purely on rank or the actual department structure? I think information like "officer X reports directly to Y" could be hard for outside observers to document.

**Comment by tomx4096** [comment]
@dismantl i was working on this but was unsure about where the data would come from (i ended up with the above PR). namely, it could be hard for an outsider to observe and document "officer X reports to Y". can it be in FOIA reports? some departments have the higher ranks listed online (http://assets.lapdonline.org/assets/pdf/new_org_chart.pdf) but not the entire roster hierarchy.

**Comment by dismantl** [comment]
@tomx4096 yeah I actually got some of that data in a FOIA response for the Baltimore PD roster. Many, but not all, of the officer records list the ID of their supervisor, which could be used to build a partial hierarchy of the department (though unsure how disjointed it would be due to not all officers having this listed). But there currently isn't a data field in OO for a supervisor ID, so that would need to be added, probably to the `assignments` table I would think.

**Comment by dismantl** [comment]
@tomx4096 you can take a look at the [roster I got for BPD](https://cdn.muckrock.com/foia_files/2018/05/11/Active_employees_as_of_May_3_2018.xlsx). The column `EMPLID` shows city employee ID number, and then the `Supv ID` column indicates the employee ID number of their supervisor, if any. So a graph could potentially be built with that.

**Comment by tomx4096** [comment]
@dismantl thanks for the info and roster. i started a 2nd PR (#541 ) for an actual graph visualization

**Comment by ejfox** [comment]
@tomx4096 @redshiftzero Your PR (https://github.com/lucyparsons/OpenOversight/pull/541) looks awesome to me. Is there any remaining work needed to bring it over? Would love to help bring this feature to the finish line if I can help in any way.

**Comment by tomx4096** [comment]
> @tomx4096 @redshiftzero Your PR (#541) looks awesome to me. Is there any remaining work needed to bring it over? Would love to help bring this feature to the finish line if I can help in any way.

thanks :) it worked fine when I tested it on the Baltimore data, but that was quite some time ago. performance could be much better (it takes more than a few seconds to load the visualization) but I'm not sure how to improve it.

**Comment by sibrahi5** [comment]
Hey @ejfox for right now, we mostly would like to add some indication to the user that the data is being loaded(some intermediary screen to let the user know that data is being loaded so they aren't kept wondering), and to fix the merge conflicts. If these could be resolved, I think we should approve the PR & worry about performance in another issue/PR
